waku/v2/peermanager/connection_gater.go

```diff
@@ -103,7 +103,7 @@ func (c *ConnectionGater) validateInboundConn(addr multiaddr.Multiaddr) bool {
 	c.Lock()
 	defer c.Unlock()
 
-	if currConnections := c.limiter[ip.String()]; currConnections+1 > maxConnsPerIP {
+	if currConnections := c.limiter[ip.String()]; c.maxConnsPerIP > 0 && currConnections+1 > c.maxConnsPerIP {
 		return false
```

Comment:
Wondering if we ought to add some info/debug log here to indicate we are limiting connections as per config.